src/xmipp/libraries/reconstruction/subtract_projection.cpp

```diff
@@ -84,287 +86,283 @@
-		DIRECT_MULTIDIM_ELEM(m(),n)=(std::abs(DIRECT_MULTIDIM_ELEM(m(),n)>0.5)) ? 1:0;
-	for (int n=0; n<10; ++n) {
-		closing2D(m(), m(), 4, 0, 16);
+ double ProgSubtractProjection::evaluateFitting(const MultidimArray< std::complex<double> > &y,
+                                                const MultidimArray< std::complex<double> > &yp) const{
+	double sumY = 0;
+	double sumY2 = 0;
+	double sumE2 = 0;
+	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(y) {
+		double realyn = real(DIRECT_MULTIDIM_ELEM(y, n)); 
+		double imagyn = imag(DIRECT_MULTIDIM_ELEM(y, n)); 
+		double ereal = realyn - real(DIRECT_MULTIDIM_ELEM(yp, n));
+		double eimag = imagyn - imag(DIRECT_MULTIDIM_ELEM(yp, n));  
+		sumE2 += ereal*ereal + eimag*eimag;
+		sumY += realyn + imagyn;
+		sumY2 += realyn*realyn + imagyn*imagyn;
 	}
- 	return m;
+	auto meanY = sumY / (2.0*(double)MULTIDIM_SIZE(y));
+	auto varY = sumY2 / (2.0*(double)MULTIDIM_SIZE(y)) - meanY*meanY;
+	auto R2 = 1.0 - (sumE2/(2.0*(double)MULTIDIM_SIZE(y))) / varY; 
+	return R2;
  }
 
- Image<double> ProgSubtractProjection::binarizeMask(Projection &m) const{
- 	double maxMaskVol;
- 	double minMaskVol;
- 	m().computeDoubleMinMax(minMaskVol, maxMaskVol);
- 	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(m())
- 		DIRECT_MULTIDIM_ELEM(m(),n) = (DIRECT_MULTIDIM_ELEM(m(),n)>0.05*maxMaskVol) ? 1:0;
- 	return m;
- }
-
- Image<double> invertMask(const Image<double> &m) {
-	Image<double> PmaskI = m;
-	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(PmaskI())
-		DIRECT_MULTIDIM_ELEM(PmaskI,n) = (DIRECT_MULTIDIM_ELEM(PmaskI,n)*(-1))+1;
-	return PmaskI;
- }
-
- Image<double> subtraction(Image<double> &I1, const Image<double> &I2,
-		const Image<double> &minv, const Image<double> &m, bool subAll){
-	 if (subAll){
-		 FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(I1())
-			DIRECT_MULTIDIM_ELEM(I1,n) = DIRECT_MULTIDIM_ELEM(I1,n)-(DIRECT_MULTIDIM_ELEM(I2,n)*DIRECT_MULTIDIM_ELEM(minv,n));
-	 }
-	 else{
-		 FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(I1())
-			DIRECT_MULTIDIM_ELEM(I1,n) = (DIRECT_MULTIDIM_ELEM(I1,n)-(DIRECT_MULTIDIM_ELEM(I2,n)*DIRECT_MULTIDIM_ELEM(minv,n)))*DIRECT_MULTIDIM_ELEM(m,n);
-	 }
-
-	return I1;
- }
-
- void ProgSubtractProjection::writeParticle(const int &ix, Image<double> &img) {
-	FileName out = formatString("%d@%s.mrcs", ix, fnOut.c_str());
-	img.write(out);
-	mdParticles.setValue(MDL_IMAGE, out, ix);
- }
+double ProgSubtractProjection::checkBestModel(MultidimArray< std::complex<double> > &PFourierf, const MultidimArray< std::complex<double> > &PFourierf0,
+ const MultidimArray< std::complex<double> > &PFourierf1, const MultidimArray< std::complex<double> > &IFourierf) const { 
+	// Compute R2 coefficient for order 0 model (R20) and order 1 model (R21)
+	auto N = 2.0*(double)MULTIDIM_SIZE(PFourierf);
+	double R20 = evaluateFitting(IFourierf, PFourierf0); 
+	double R20adj = 1.0 - (1.0 - R20) * (N - 1.0) / (N - 1.0);
+	double R21 = evaluateFitting(IFourierf, PFourierf1); 
+	double R21adj = 1.0 - (1.0 - R21) * (N - 1.0) / (N - 1.0 - 1.0);
```

Comment:
yes, we have two explanatory variables so p = 2, we indicated it like this to conserve the paralelism with R20 adjusted, but as I have removed it as you indicated (because R2adj = R2 always in order 0), I have changed by -2 instead of -1 -1 that doesn´t look like a bug as before